### PR TITLE
I2b2tm-182 Auth0 Login provider fix

### PR DESCRIPTION
**Pull request by gaborkorodi-hms**
Add a flexibility to the Auth0 Plugin that uses the `provider` field of the Auth0 User Profile to validate and record newly added user records.

The validation of a first-time user, after she has been added by the admin, is based on comparing Auth0 User Profile (generated by Auth0 authentication) vs. what the admin is entered when setting up the user.

### CODE REVIEW

File: grails-app/services/org/transmart/plugin/auth0/Auth0Service.groovy [hunk 1]
```diff
@@ -132,13 +143,81 @@ class Auth0Service implements InitializingBean {
...
+        // Hopefully, if no user found or more than one user found, this will return NULL?!
+    }
+
+    /**
+     * Determine if the user is logging in for the first time, or not. Based on the
+     * initial user record creation by the admin, the uniqueId is either set to
+     * CONNECTION|CONNECTION-ID (email, ORCiD or other known identifier) and name is 'unregistered'
+     * @param userInfo the full Auth0 profile for the user that was returned by the /userinfo endpoint
+     */
+    AuthUser getUninitedUser(JSONObject userInfo) {
```
Comment (alexnktn) [inline grails-app/services/org/transmart/plugin/auth0/Auth0Service.groovy]: Appears to be a typo here.

File: grails-app/services/org/transmart/plugin/auth0/Auth0Service.groovy [hunk 2]
```diff
@@ -189,105 +268,71 @@ class Auth0Service implements InitializingBean {
...
-			}
-		}
-
+		logger.info 'createCredentials() Auth0 userinfo: {}', userInfo
+
+        // Create a base `Credentials` object, with reasonable defaults.
+        Credentials credentials = new Credentials(
+                accessToken: accessToken,
+                connection: userInfo.getJSONArray('identities').getJSONObject(0).getString('connection'),
+                email: userInfo.optString('email'),
```
Comment (alexnktn) [inline grails-app/services/org/transmart/plugin/auth0/Auth0Service.groovy]: email toLowerCase();

Comment (gaborkorodi-hms) [reply grails-app/services/org/transmart/plugin/auth0/Auth0Service.groovy]: no need, will do later

File: grails-app/services/org/transmart/plugin/auth0/Auth0Service.groovy (hunk 1 above)
Comment (gaborkorodi-hms) [reply grails-app/services/org/transmart/plugin/auth0/Auth0Service.groovy]: No longer used, using proper english now

**Comment by gaborkorodi-hms** [comment]
Please review the latest changes. Test results are being uploaded and believe logging and comments are sufficient for release.

**Comment by alexnktn** [comment]
looks ok, need to be tested in AWS

**Pull request merged**